master/internal/plugin/saml/service.go

```diff
@@ -227,10 +236,25 @@ func (s *Service) toUserAttributes(response *saml.Assertion) *userAttributes {
 		return nil
 	}
 
+	strAgentUID := getSAMLAttribute(response, string(s.userConfig.agentUIDAttributeName))
+	tempAgentUID, err := strconv.Atoi(strAgentUID)
+	if err != nil {
+		logrus.WithError(err).WithField("agentUID", strAgentUID).Error("unable to convert to integer")
```

Comment:
I think this error should be very rare / surprising but we should log it just in case.

Reply:
wdy think about having this parser return an err? the only place it's used does support returning an error that might be the right behavior here.

Reply:
Great idea, thanks!